common/utils/budget-sums.test.js

```diff
@@ -3679,6 +3687,461 @@ describe('budget summing methods', () => {
+            }
+          },
+          expenses: {
+            2023: {
+              total: 1550000,
+              federal: 619308,
+              medicaid: 825744,
+              state: 206436
+            },
+            total: {
+              total: 1550000,
+              federal: 619308,
+              medicaid: 825744,
+              state: 206436
+            }
+          },
+          combined: {
+            2023: {
+              total: 6709662,
+              federal: 3865557,
+              medicaid: 5154076,
+              state: 1288519
+            },
+            total: {
+              total: 6709662,
+              federal: 3865557,
+              medicaid: 5154076,
+              state: 1288519
+            }
+          }
+        }
+      };
+      const actual = sumShareCostsForFundingCategory({
+        activityTotals,
+        budget,
+        costCategoryShare,
+        fundingCategory,
+        year
+      });
+      expect(actual).toEqual(expected);
+
+      // Below are the nitty gritty calculation checks
+
+      // statePersonnel section -------
+
+      // Amounts to add to the cost category
+      const statePersonnelTotalToAdd =
+        activityTotals.data.statePersonnel['2023'];
+      const statePersonnelMedicaidShareToAdd =
+        costCategoryShare.medicaidShare.statePersonnel;
+      const statePersonnelFedShareToAdd =
+        costCategoryShare.fedShare.statePersonnel;
+      const statePersonnelStateShareToAdd =
+        costCategoryShare.stateShare.statePersonnel;
+
+      // Amounts for the cost category (CC)'s year
+      const statePersonnelOldCCYearTotal =
+        simplifiedBudgetProp.mando.statePersonnel['2023'].total;
+      const statePersonnelOldCCYearMedicaid =
+        simplifiedBudgetProp.mando.statePersonnel['2023'].medicaid;
+      const statePersonnelOldCCYearFederal =
+        simplifiedBudgetProp.mando.statePersonnel['2023'].federal;
+      const statePersonnelOldCCYearState =
+        simplifiedBudgetProp.mando.statePersonnel['2023'].state;
+      const statePersonnelNewCCYearTotal =
+        actual.mando.statePersonnel['2023'].total;
+      const statePersonnelNewCCYearMedicaid =
+        actual.mando.statePersonnel['2023'].medicaid;
+      const statePersonnelNewCCYearFederal =
+        actual.mando.statePersonnel['2023'].federal;
+      const statePersonnelNewCCYearState =
+        actual.mando.statePersonnel['2023'].state;
+
+      const expectedStatePersonnelCCYearTotal =
+        statePersonnelOldCCYearTotal + statePersonnelTotalToAdd;
+      const expectedStatePersonnelCCYearMedicaid =
+        statePersonnelOldCCYearMedicaid + statePersonnelMedicaidShareToAdd;
+      const expectedStatePersonnelCCYearFederal =
+        statePersonnelOldCCYearFederal + statePersonnelFedShareToAdd;
+      const expectedStatePersonnelCCYearState =
+        statePersonnelOldCCYearState + statePersonnelStateShareToAdd;
+
+      expect(statePersonnelNewCCYearTotal).toEqual(
```

Comment:
could we do this with just the actual values and not make the constants first? it might make it more readable

Reply:
I was trying to label everything purely as a means to help make the calculation more understandable, so if it's not serving that purpose, I agree that it should change!

I'll cut out all the "expected" variables here and see if that helps. If it's still too much, let me know and I'll re-edit.